### ISSUE TITLE
Should a talkback also allow sending data, not just requesting it

### ISSUE THREAD

**Issue by RebeccaStevens** [tim-smart/strict-callbag#1]
There should be different types of talkbacks, pullable ones and ones that push.
The push ones need to be able to send data but currently talkbacks only support requesting it.

**Comment by tim-smart** [comment]
I have implemented some duplex streaming with callbags, and I wouldn't recommend using the talkback to send data.

You lose back-pressure support amongst other things.

A pattern that seems to work well is accepting an outbound source as an argument to your factory function. I.e.

```typescript
type WebsocketError = { _tag: "error"; cause: unknown }

function createWebsocket(
  url: string,
  outbound: Source<string, never>,
): Source<string, WebsocketError> {
  // TODO: Implement websocket source from arguments
  return (_, sink) => {}
}
```

So a duplex can be represented as a function that takes the outbound stream, and returns an inbound stream (`outbound -> inbound`)

**Comment by tim-smart** [comment]
Another option is to return a `Source` & `Sink` from the factory:

```typescript
type WebsocketError = { _tag: "error"; cause: unknown }

function createWebsocket(
  url: string,
): readonly [Source<string, WebsocketError>, Sink<string, WebsocketError>] {
  // TODO: Implement websocket source from arguments
  const source: Source<string, WebsocketError> = (_, sink) => {}
  const sink: Sink<string, WebsocketError> = (signal, data) => {
    // TODO implement sink
  }

  return [source, sink]
}
```

But I don't like this option as much :)

**Comment by RebeccaStevens** [comment]
@tim-smart Do you have a discord something where we could dm?
I'm kind of new to callbags (though I have know of them for a long time and played with them in the past) and was wondering if you could take a look at my implementation and give me feedback.

**Comment by tim-smart** [comment]
Yeah sure, my Discord username: timsmart#8313
